code/modules/projectiles/projectile_parent.dm

```diff
@@ -632,6 +696,11 @@ ABSTRACT_TYPE(/datum/projectile)
 	/// for on_pre_hit. Causes it to early-return TRUE if the thing checked was already cleared for pass-thru
 	var/atom/last_thing_hit
 
+	/// Set to TRUE if you want particles to spawn when you hit a non living thing
+	var/has_impact_particles = FALSE
+	/// Set to TRUE if you want kinetic impact particles, FALSE assumes energy projectiles
+	var/kinetic_impact = TRUE
```

Comment:
Shouldn't this just use damage types or typechecks? At least it should default to using them, with this var existing as an override if needed.

Reply:
It now checks for any damage that is D_ENERGY, D_BURNING, D_RADIOACTIVE or D_TOXIC, and if not overriden spawns the appropriate particle.